templates/CRM/Core/BillingBlock.js

```diff
@@ -58,31 +60,20 @@
   function civicrm_billingblock_set_card_type(ccnumber) {
     // Based on http://davidwalsh.name/validate-credit-cards
     // See also https://en.wikipedia.org/wiki/Credit_card_numbers
+    // @todo these patterns should be part of the credit card option group, instead of hard coded
     var card_types = {
-      'mastercard': '(5[1-5][0-9]{2}|2[3-6][0-9]{2}|22[3-9][0-9]|222[1-9]|27[0-1][0-9]|2720)[0-9]{12}',
```

Comment:
not sure we should be changing this - we should be running off the lower case 'name' field?